### ISSUE TITLE
About custom components on RichText mode.

### ISSUE THREAD

**Issue by nio-p** [mdx-editor/editor#73]
Hello, 
Thank you for developing great software.

I'm currently use mdx-editor to preview the markdown contents.
But I can not preview my custom components such as Heading, List components.

I would like to show my custom Heading on RitchText mode like below.

```
<MDXEditor
    components={
      h1: ({ children }) => <H1>{children}</H1>,
      h2: ({ children }) => <H2>{children}</H2>,
      ...
    }
/>
```

This similars to ReactMarkdown's syntax. 

Is it possible to do something like this?

or...

```
<MDXEditor
    editorRef={ref}
    diffSourcePlugin({ viewMode: 'source' }),
/>
```

In this case, `ref.current.?.getMarkdown() ?? ''` returns always empty string.
`viewMode: source` is nice, but getMarkdown doesn't work.
I can preview by using other system like the ReactMarkdown If I can get markdown content. 

If either of these is possible, my problem could be solved.

Thanks,

**Comment by petyosi** [comment]
Hi, 

The first option is certainly possible, but something you probably don't want to go into, because you can't use your custom components directly. Instead, you should build **editor nodes** that look like them. This is quite involved, and uses the Lexical API. I am not sure what your custom H1 and Lists, but I would recommend styling the default editor elements with CSS instead. If you decide to try injecting your custom components, you can take a look at the source code of the editor, the headings plugin, to be more precise. Again, I don't think that it is worth it. 

The second approach you outline is actually a bug which I just fixed in 0.22.4. [Take a look at the sandbox](https://codesandbox.io/s/goofy-julien-wj63m9?file=/package.json) and let me know if I am missing something. 

Regards,

**Comment by nio-p** [comment]
@petyosi 

Hello, 
0.22.4 worked fine! 
It solved my problems !

I appreciate it very much !! 👍 👍 

Thanks,
